### ISSUE TITLE
HeatSolveVec: DG and Periodic Boundaries

### ISSUE THREAD

**Issue by ecs-kev** [ElmerCSC/elmerfem#287]
Hi,

I recently stumbled upon this forum post (http://www.elmerfem.org/forum/viewtopic.php?f=3&t=7355) as I wanted to model a case with multiple discontinous boundaries and discovered the HeatSolveVec solver.

I mostly managed to get it to work, except when I try to run a model with periodic boundaries while Discontinous Galerkin is set to True. Is this a capability that isn't supported by the HeatSolveVec solver yet? Using one without the other works as expected.

Not sure if this is the right place for this question. I meant to post this in the elmerfem forum, but for some reason I don't seem to receive the account activation email after registering. 

Kind regards
Kevin

**Comment by raback** [comment]
Yes, it seems that this is unfortunately not well supported. Sorry for that!

The discontinuity over boundaries was something needed by a single project. Hence initially there was not effort to make it compatible with all other possible library features. The challenge lies in the numbering. The periodic/conforming/mortar BCs have been mainly written for nodal dofs. In fact for DG you would have issue from which node to inherit the value as there are physical nodes with several possible values. If you want to work on this in ernest maybe we could find some strategy.

**Comment by ecs-kev** [comment]
No problem, thanks for letting me know that those 2 boundary conditions currently don't work together.
For now, I can get around this limitation by modelling the full 360° geometry or replacing the periodic BC with 0 Heatflux BCs.

I might revisit this in a few months time, though I don' have much experience coding in Fortran, so I'll have to see how realistic it would be for me to implement this.
